src/WingmanDataTrail/ListControls/MetricsSorter/MetricsSorter.tsx

```diff
@@ -144,8 +145,17 @@ export class MetricsSorter extends SceneObjectBase<MetricsSorterState> {
     return this.usageFetcher.getUsageForMetric(metricName, usageType);
   }
 
+  public getUsageDetailsForMetric(metricName: string, usageType: MetricUsageType): Promise<MetricUsageDetails> {
+    return this.usageFetcher.getUsageDetailsForMetric(metricName, usageType);
+  }
+
+  // Converts MetricUsageDetails format to simple counts (Record<string, number>) for backward compatibility with sorting logic
   public getUsageMetrics(usageType: MetricUsageType): Promise<Record<string, number>> {
-    return this.usageFetcher.getUsageMetrics(usageType);
+    return this.usageFetcher.getUsageMetrics(usageType).then((metrics) => {
+      return Object.fromEntries(
+        Object.entries(metrics).map(([metricName, metricUsage]) => [metricName, metricUsage.count])
+      );
```

Comment:
What do you think about this simpler version:
```suggestion
      const metricsToCounts: Record<string, number> = {};
      for (const metric in metrics) {
        metricsToCounts[metric] = metrics[metric].count;
      }
      return metricsToCounts;
```

(this way we don't need to convert `metrics` from an object to entries and back + an additional `map`)